src/longbow/peek/command.py

```diff
@@ -0,0 +1,360 @@
+            logger.info("Detecting best model in %d reads", min(read_count, num_reads))
+
+    # Create queues for data:
+    queue_size = threads * 2 if threads < 10 else 20
+    manager = mp.Manager()
+    input_data_queue = manager.Queue(maxsize=queue_size)
+    results = manager.Queue()
+
+    # Start worker sub-processes:
+    worker_pool = []
+
+    for i in range(threads):
+        p = mp.Process(
+            target=_worker_attempt_segmentations_fn, args=(input_data_queue, results, i, min_length, max_length, min_rq, models)
+        )
+        p.start()
+        worker_pool.append(p)
+
+    pysam.set_verbosity(0)  # silence message about the .bai file not being found
+    with pysam.AlignmentFile(
+        input_bam if start_offset == 0 else input_bam.name, "rb", check_sq=False, require_index=False
+    ) as bam_file:
+
+        # If we're chunking, advance to the specified virtual file offset.
+        if start_offset > 0:
+            bam_file.seek(start_offset)
+
+        # Start output worker:
+        res = manager.dict()
+        output_worker = mp.Process(
+            target=_collect_thread_fn,
+            args=(results, output_model, not sys.stdin.isatty(), res, min(read_count, num_reads))
+        )
+        output_worker.start()
+
+        # Add in a sentinel value at the end of the queue - one for each subprocess - so we guarantee
+        # that all subprocesses will exit:
+        reads_seen = 0
+        iter_data = itertools.chain(bam_file, (None,) * threads)
+        for r in iter_data:
+            # We have to adjust for our sentinel value if we've got to it:
+            if r is not None:
+                r = r.to_string()
+            input_data_queue.put(r)
+
+            if start_offset > 0:
+                if bam_file.tell() >= end_offset or r is None:
+                    [input_data_queue.put(None) for _ in range(threads)]
+                    break
+
+            if reads_seen >= num_reads:
+                [input_data_queue.put(None) for _ in range(threads)]
+                break
+
+            # Increment reads seen
+            reads_seen += 1
+
+        logger.debug("Finished reading data and sending it to sub-processes.")
+        logger.debug("Waiting for sub-processes to finish...")
+
+        # Wait for our input jobs to finish:
+        for p in worker_pool:
+            p.join()
+
+        logger.debug("All workers stopped.")
+        logger.debug("Terminating output process.")
+
+        # Now that our input processes are done, we can add our exit sentinel onto the output queue and
+        # wait for that process to end:
+        results.put(None)
+        output_worker.join()
+
+    logger.info("Histogram of most likely model counts:")
+
+    res = dict(sorted(res.items(), key=lambda item: item[1], reverse=True))
+    plot_model_counts(res)
+    best_model = next(iter(res.keys()))
+
+    logger.info(f"Overall most likely model: {best_model}")
```

Comment:
Can you add a statistic to go with this log statement?  Seems like you can pull it directly from the histogram.

Reply:
Added.